src/main/java/org/snomed/snowstorm/core/data/services/BranchReviewService.java

```diff
@@ -228,6 +231,106 @@ public void applyMergeReview(MergeReview mergeReview) throws ServiceException {
+	}
+
+	private Set<Description> mergePublishedDescriptions(Set<Description> sourceDescriptions, Set<Description> targetDescriptions) {
+		Map<String, Description> mergedDescriptions = new HashMap<>();
+
+		for (Description sourceDescription : sourceDescriptions) {
+			boolean publishedAndUnchanged = sourceDescription.getReleasedEffectiveTime() != null && sourceDescription.getEffectiveTimeI() != null;
+			if (publishedAndUnchanged) {
+				mergedDescriptions.put(sourceDescription.getId(), sourceDescription);
+			}
+		}
+
+		for (Description targetDescription : targetDescriptions) {
+			String targetDescriptionId = targetDescription.getId();
+			Description sourceDescription = mergedDescriptions.get(targetDescriptionId);
+
+			if (sourceDescription == null) {
+				mergedDescriptions.put(targetDescriptionId, targetDescription);
+			} else if (!sourceDescription.isReleasedMoreRecentlyThan(targetDescription)) {
+				mergedDescriptions.put(targetDescriptionId, targetDescription);
+			}
+		}
+
+		return new HashSet<>(mergedDescriptions.values());
+	}
+
+	private Set<Relationship> mergePublishedRelationships(Set<Relationship> sourceRelationships, Set<Relationship> targetRelationships) {
+		Map<String, Relationship> mergedRelationships = new HashMap<>();
+
+		for (Relationship sourceRelationship : sourceRelationships) {
+			boolean publishedAndUnchanged = sourceRelationship.getReleasedEffectiveTime() != null && sourceRelationship.getEffectiveTimeI() != null;
+			if (publishedAndUnchanged) {
+				mergedRelationships.put(sourceRelationship.getId(), sourceRelationship);
+			}
+		}
+
+		for (Relationship targetRelationship : targetRelationships) {
+			String targetComponentId = targetRelationship.getId();
+			Relationship sourceRelationship = mergedRelationships.get(targetComponentId);
+
+			if (sourceRelationship == null) {
+				mergedRelationships.put(targetComponentId, targetRelationship);
+			} else if (!sourceRelationship.isReleasedMoreRecentlyThan(targetRelationship)) {
+				mergedRelationships.put(targetComponentId, targetRelationship);
+			}
+		}
+
+		return new HashSet<>(mergedRelationships.values());
+	}
+
+
+	private Set<Axiom> mergePublishedAxioms(Set<Axiom> sourceAxioms, Set<Axiom> targetAxioms) {
+		Map<String, Axiom> mergedAxioms = new HashMap<>();
+
+		for (Axiom sourceAxiom : sourceAxioms) {
+			if (sourceAxiom.isReleased()) {
+				mergedAxioms.put(sourceAxiom.getAxiomId(), sourceAxiom);
+			}
+		}
+
+		for (Axiom targetAxiom : targetAxioms) {
+			String targetAxiomId = targetAxiom.getAxiomId();
+			Axiom sourceAxiom = mergedAxioms.get(targetAxiomId);
+
+			if (sourceAxiom == null) {
+				mergedAxioms.put(targetAxiomId, targetAxiom);
+			} else {
+				ReferenceSetMember sourceMember = sourceAxiom.getReferenceSetMember();
+				ReferenceSetMember targetMember = targetAxiom.getReferenceSetMember();
+
+				if (!sourceMember.isReleasedMoreRecentlyThan(targetMember)) {
+					mergedAxioms.put(targetAxiomId, targetAxiom);
+				}
+			}
+		}
```

Comment:
@dmcgihtsdo  We can create one method that uses snomed components as arguments to do merge for descriptions, relationships and axioms.